CPP_STYLE_GUIDE.md

```diff
@@ -257,6 +258,27 @@ class ExampleClass {
 };
 ```
 
+### When to use AliasedBuffers
+
+When composite objects, specifically JS arrays (that have backing
+native buffers), are operated globally (modifiable outside of the
+lexical scope) or out-of-band (modifiable outside of the callsite)
+the observability of such changes are lost from Virtual Machine layer.
+So if the object's backing data is being modified from C++ side
+out-of-band with the calling context, the recommended abstraction to use
+is AliasedBuffer class, that has place holders to hook observers
+for data mutations.
+
+```c++
+// Create an AliasedBuffer
+AliasedBuffer<uint32_t, v8::Uint32Array> data;
+...
+
+// Modify the data through the setters at specified index
+// Thus, the setter has a chance to hook mutation observers
+data.SetValue(0, 12345);
```

Comment:
We usually use the `[]` accessors, so I think the style guide should list them as the primary alternative?